apps/microsoft-teams/app-actions/src/actions/handle-app-event.ts

```diff
@@ -0,0 +1,48 @@
+import { AppActionCallContext } from '@contentful/node-apps-toolkit';
+import { AppActionCallResponse, EntryActivityMessage, Topic } from '../types';
+import { EntryProps } from 'contentful-management/types';
+import helpers from '../helpers';
+
+interface AppActionCallParameters {
+  payload: string;
+  topic: string;
+  eventDatetime: string;
+}
+
+export const handler = async (
+  parameters: AppActionCallParameters,
+  context: AppActionCallContext
+): Promise<AppActionCallResponse<EntryActivityMessage>> => {
+  const {
+    cma,
+    appActionCallContext: { appInstallationId },
+  } = context;
+
+  const { payload, topic: topicString, eventDatetime } = parameters;
+
+  // TODO parse entry and topic
+  const entry = JSON.parse(payload) as EntryProps;
+  const topic = topicString as Topic;
+
+  const entryActivity = await helpers.buildEntryActivity({ entry, topic, eventDatetime }, cma);
+
+  // TODO check app config to see if there are any subscriptions matching, return if none
+  const appInstallation = await cma.appInstallation.get({ appDefinitionId: appInstallationId });
+  console.log(appInstallation);
```

Comment:
Remove `console.log`?

Reply:
it's just a way for me to leave that line above in place without the linter complaining. 😄  will remove it next round.